hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointManagerImpl.java

```diff
@@ -122,7 +122,7 @@ public void removeEndpoint(final ClientEndpoint ce, boolean closeImmediately) {
         final Connection connection = endpoint.getConnection();
         if (closeImmediately) {
             try {
-                connection.close();
+                connection.close(null, null);
```

Comment:
Hmm, it would be nice to have a reason here as well. At least we could log if the connection was closed immediately or not (`ClientDisconnectionOperation` vs. authentification failure and `DestroyEndpointTask`). For more detailed logging we may need an additional parameter here as well :(